edxval/tests/test_api.py

```diff
@@ -269,6 +270,191 @@ def test_get_url_for_profile(self):
         self.assertEqual(url, u'http://www.meowmix.com')
 
 
+class GetVideoForCourseProfiles(TestCase):
+    """Tests get_video_info_for_course_and_profiles in api.py"""
+
+    def setUp(self):
+        """
+        Creates two courses for testing
+
+        Creates two videos with 2 encoded videos for the first course, and then
+        2 videos with 1 encoded video for the second course.
+        """
+        mobile_profile = Profile.objects.create(**constants.PROFILE_DICT_MOBILE)
+        desktop_profile = Profile.objects.create(**constants.PROFILE_DICT_DESKTOP)
+
+        self.course_id = 'test-course'
+        # 1st video
+        video = Video.objects.create(**constants.VIDEO_DICT_FISH)
+        EncodedVideo.objects.create(
+            video=video,
+            profile=mobile_profile,
+            **constants.ENCODED_VIDEO_DICT_MOBILE
+        )
+        EncodedVideo.objects.create(
+            video=video,
+            profile=desktop_profile,
+            **constants.ENCODED_VIDEO_DICT_DESKTOP
+        )
+        CourseVideo.objects.create(video=video, course_id=self.course_id)
+        # 2nd video
+        video = Video.objects.create(**constants.VIDEO_DICT_STAR)
+        EncodedVideo.objects.create(
+            video=video,
+            profile=mobile_profile,
+            **constants.ENCODED_VIDEO_DICT_MOBILE2
+        )
+        EncodedVideo.objects.create(
+            video=video,
+            profile=desktop_profile,
+            **constants.ENCODED_VIDEO_DICT_DESKTOP2
+        )
+        CourseVideo.objects.create(video=video, course_id=self.course_id)
+
+        self.course_id2 = "test-course2"
+        # 3rd video different course
+        video = Video.objects.create(**constants.VIDEO_DICT_TREE)
+        EncodedVideo.objects.create(
+            video=video,
+            profile=mobile_profile,
+            **constants.ENCODED_VIDEO_DICT_MOBILE3
+        )
+        CourseVideo.objects.create(video=video, course_id=self.course_id2)
+
+        # 4th video different course
+        video = Video.objects.create(**constants.VIDEO_DICT_PLANT)
+        EncodedVideo.objects.create(
+            video=video,
+            profile=desktop_profile,
+            **constants.ENCODED_VIDEO_DICT_DESKTOP3
+        )
+        CourseVideo.objects.create(video=video, course_id=self.course_id2)
+
+    def _create_video_dict(self, video, profiles, encoded_videos):
```

Comment:
Calls to this would be clearer if it took a dictionary from profile name to encoded video instead taking profiles and encoded_videos separately as lists.